src/odemis/acq/stream/_projection.py

```diff
@@ -1083,11 +1112,19 @@ class NeedRecomputeException(Exception):
         need_recompute = True
         while need_recompute:
             z = self._zFromMpp()
-            rect = self._rectWorldToPixel(self.rect.value)
-            # convert the rect coords to tile indexes
-            rect = [l / (2 ** z) for l in rect]
-            rect = [int(math.floor(l / das.tile_shape[0])) for l in rect]
-            x1, y1, x2, y2 = rect
+
+            # calculate the image pixels inside the view (rect)
+            view_rect = self._rectWorldToPixel(self.rect.value)
```

Comment:
I think to support rotation properly, you'll need to adjust _rectWorldToPixel(). Instead of trying to itself compute the coordinates, it should just compute the coordinates of the 4 corners of .rect, and then pass them to Stream.getPixelCoordinates(). This then returns 4 points in the "world" referential, you can then get the bounding box of these 4 points.

Reply:
I'm not planning to support rotation properly at this time, because it's going to require too much time and these images are usually 0 degrees or minimially rotated. passing the corners of the rect to getPixelCoordinates returns None, likely because of a -1 difference or rounding differences again. We also need to rotate the tile index calculation so its going to be a a fair chunk of work that isn't a priority right now.

Reply:
I see. I actually had in mind the SEM overview images that would have 180° rotation... but I realise that for now this images are not stored in pyramidal format (as it done in TFS Maps typically). So we don't need to worry (at least until Odemis also supports SEM overview acquisition!)

The issue with `getPixelCoordinates()` returning `None` is... "interesting". I hadn't thought of it, but indeed that the danger of passing the coordinates just at the border of the data. We might need to break this function into a lower-level one which only does referential conversion, without checking the shape of the data.